src/Nethermind/Nethermind.Init/Steps/InitializeNetwork.cs

```diff
@@ -478,19 +480,29 @@ private async Task InitPeer()
             {
                 await plugin.InitNetworkProtocol();
             }
-
-            PeerLoader peerLoader = new(_networkConfig, _api.NodeStatsManager, peerStorage, _api.LogManager);
+            
+            NodesLoader nodesLoader = new(_networkConfig, _api.NodeStatsManager, peerStorage, _api.RlpxPeer, _api.LogManager);
+            EnrDiscovery enrDiscovery = new (); // initialize with a proper network
+            CompositeNodeSource nodeSources = new(_api.StaticNodesManager, nodesLoader, enrDiscovery, _api.DiscoveryApp);
+            _api.PeerPool = new PeerPool(nodeSources, _api.NodeStatsManager, peerStorage, _networkConfig, _api.LogManager);
             _api.PeerManager = new PeerManager(
                 _api.RlpxPeer,
-                _api.DiscoveryApp,
+                _api.PeerPool,
                 _api.NodeStatsManager,
-                peerStorage,
-                peerLoader,
                 _networkConfig,
-                _api.LogManager,
-                _api.StaticNodesManager);
-            
-            _api.PeerManager.Init();
+                _api.LogManager);
+
+            string chainName = ChainId.GetChainName(_api.ChainSpec!.ChainId).ToLowerInvariant();
+#pragma warning disable CS4014
+            enrDiscovery.SearchTree($"all.{chainName}.ethdisco.net").ContinueWith(t =>
+#pragma warning restore CS4014
+            {
+                if (t.IsFaulted)
+                {
+                    _logger.Error($"ENR discovery failed: {t.Exception}");
+                    
+                }
+            });
```

Comment:
Should we do a proper close of this task on shutdown?

Reply:
no need, it will just stop when the main thread dies and it has no part-finished work that would be any risky